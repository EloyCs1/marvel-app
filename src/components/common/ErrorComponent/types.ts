export interface ErrorComponentProps {
  testId?: string;
  tryAgain?: () => void;
}
