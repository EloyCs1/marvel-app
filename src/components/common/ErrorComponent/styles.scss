.errorComponent {
  &__title {
    text-transform: uppercase;
    font-size: 32px;
    font-weight: 700;
    line-height: normal;
    position: relative;
    margin-top: 48px;
    margin-bottom: 48px;

    &::before {
      left: 50px;
      transform-origin: left bottom;
      background-color: var(--color-red);
      content: "";
      display: block;
      height: 2px;
      position: absolute;
      transform: rotate(-45deg) skewX(45deg);
      width: 20px;
    }

    &::after {
      left: -6px;
      transform-origin: right top;
      background-color: var(--color-red);
      content: "";
      display: block;
      height: 2px;
      position: absolute;
      transform: rotate(-45deg) skewX(45deg);
      width: 20px;
    }
  }

  &__subTitle {
    text-transform: uppercase;
    font-size: 24px;
    font-weight: 700;
    line-height: normal;
  }

  a {
    text-decoration: none;
    color: inherit;
    cursor: pointer;
    &:hover {
      color: #eb2328;
    }
  }
}
