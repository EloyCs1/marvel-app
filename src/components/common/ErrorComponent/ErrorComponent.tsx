import { useIntl } from "react-intl";

import { ErrorComponentProps } from "./types";
import "./styles.scss";

const ErrorComponent: React.FC<ErrorComponentProps> = ({ testId = "errorComponent", tryAgain }) => {
  const { formatMessage } = useIntl();
  return (
    <div className="errorComponent" data-testid={testId}>
      <p className="errorComponent__title">{formatMessage({ id: "errorComponent.title" })}</p>
      <p className="errorComponent__subTitle">{formatMessage({ id: "errorComponent.subTitle" })}</p>
      {tryAgain && (
        <a onClick={tryAgain} data-testid={`${testId}-tryAgain`}>
          {formatMessage({ id: "errorComponent.tryAgain" })}
        </a>
      )}
    </div>
  );
};
export default ErrorComponent;
