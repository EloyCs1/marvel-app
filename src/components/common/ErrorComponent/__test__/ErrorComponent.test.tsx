import { render } from "@testing-library/react";
import ErrorComponent from "../ErrorComponent";

import { AppWrapper } from "src/__test__/test.utils";

describe("ErrorComponent", () => {
  test("Default render", () => {
    const wrapper = render(
      <AppWrapper>
        <ErrorComponent />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("tryAgain render", () => {
    const tryAgain = jest.fn();
    const wrapper = render(
      <AppWrapper>
        <ErrorComponent tryAgain={tryAgain} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
