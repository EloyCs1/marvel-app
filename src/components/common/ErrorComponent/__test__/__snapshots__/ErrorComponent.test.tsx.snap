// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent Default render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="errorComponent"
        data-testid="errorComponent"
      >
        <p
          class="errorComponent__title"
        >
          Error
        </p>
        <p
          class="errorComponent__subTitle"
        >
          Upsss, algo sali贸 mal!!!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="errorComponent"
      data-testid="errorComponent"
    >
      <p
        class="errorComponent__title"
      >
        Error
      </p>
      <p
        class="errorComponent__subTitle"
      >
        Upsss, algo sali贸 mal!!!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ErrorComponent tryAgain render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="errorComponent"
        data-testid="errorComponent"
      >
        <p
          class="errorComponent__title"
        >
          Error
        </p>
        <p
          class="errorComponent__subTitle"
        >
          Upsss, algo sali贸 mal!!!
        </p>
        <a
          data-testid="errorComponent-tryAgain"
        >
          Intentar de nuevo
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="errorComponent"
      data-testid="errorComponent"
    >
      <p
        class="errorComponent__title"
      >
        Error
      </p>
      <p
        class="errorComponent__subTitle"
      >
        Upsss, algo sali贸 mal!!!
      </p>
      <a
        data-testid="errorComponent-tryAgain"
      >
        Intentar de nuevo
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
